### PR TITLE
Add `autumn generate channel` command for real-time broadcast channels

### PR DESCRIPTION
**Pull request by madmax983**
Implements a new generator subcommand that scaffolds real-time broadcast channels using Autumn's existing `Channels` API, with support for both SSE-over-htmx (default) and raw WebSocket transports.

## Summary

This adds `autumn generate channel <Name>` to scaffold a complete real-time channel implementation, including:
- Channel handler(s) that publish/subscribe through the `Channels` API
- An htmx SSE live view by default (zero client JS required) or a raw `#[ws]` WebSocket handler with `--ws`
- Automatic `main.rs` route registration and module wiring
- An in-process smoke test that verifies pub/sub delivery
- Cargo.toml dependency and feature management

## Key Changes

- **New module `autumn-cli/src/generate/channel.rs`** (~1073 lines): Core generator logic
  - `plan_channel()`: Computes file actions (create/modify) for the channel scaffold
  - `Transport` enum: Distinguishes between SSE and WebSocket transports
  - `render_sse_channel_file()`: Generates SSE handler with htmx `sse-connect`/`sse-swap` wiring
  - `render_ws_channel_file()`: Generates `#[ws]` handler with bidirectional relay logic
  - `render_smoke_test()`: Creates transport-agnostic integration test using `TestApp`
  - Comprehensive unit tests covering both transports, file plan validation, and edge cases

- **Updated `autumn-cli/src/main.rs`**: 
  - Added `Channel` variant to `GenerateCommands` enum with `--sse`, `--ws`, `--dry-run`, `--force` flags
  - Wired command parsing to `generate::channel::run()`
  - Added CLI parsing tests for the new command

- **Enhanced `autumn-cli/src/generate/schema_edit.rs`**:
  - Extracted `find_routes_body_range()` helper to locate `routes![...]` macro body
  - Added `remove_routes_entries_with_prefix()` to clean stale route entries when regenerating with a different transport (e.g., switching from SSE to WebSocket via `--force`)

- **Integration tests in `autumn-cli/tests/generate.rs`**:
  - `generate_channel_creates_all_expected_files()`: Verifies SSE scaffold completeness
  - `generate_channel_ws_emits_ws_handler()`: Validates WebSocket transport output
  - `generate_channel_force_transport_switch_does_not_strand_stale_routes()`: Regression test for transport switching
  - `generated_channel_cargo_checks()`: End-to-end cargo check (ignored by default, slow)

- **Documentation in `docs/guide/generators.md`**: Added channel generator to the generators table and detailed usage guide

- **CI workflow update in `.github/workflows/generator-conformance.yml`**: Added channel generator tests to conformance checks

## Notable Implementation Details

- **Transport switching safety**: When regenerating a channel with a different transport via `--force`, the generator removes stale route entries (e.g., `chat_page`/`chat_events` when switching from SSE to WebSocket) before adding new ones, preventing dangling references in `main.rs`

- **Smoke test design**: The test re-declares the publish handler locally (since `tests/` cannot import the binary crate) and uses `TestApp` with real `Channels` pub/sub to verify message delivery, not a stub

- **Dependency management**: Automatically enables the `"ws"` feature on `autumn-web` (required for channels, SSE, and `#[ws]`), adds `serde` for form deserialization, and conditionally adds `maud` for SSE's Markup rendering

- **SSE view**: Uses htmx's SSE extension with declarative `sse-connect`/`sse-swap` attributes — zero client JavaScript authored by the user

- **WebSocket handler**: Implements a bidirectional relay with proper shutdown handling, lagged message recovery, and logging of publish failures

https://claude.ai/code/session_019V7c1Q5gvEsix2cWQJ9m79

### CODE REVIEW

File: autumn-cli/src/generate/channel.rs [hunk 1]
```diff
@@ -0,0 +1,1073 @@
...
+    let channels = state.channels().clone();
+    let mut rx = channels.subscribe(TOPIC);
+
+    WithShutdown(
+        move |mut socket: WebSocket, shutdown: CancellationToken| async move {{
+            loop {{
+                tokio::select! {{
+                    incoming = socket.recv() => match incoming {{
+                        Some(Ok(Message::Text(text))) => {{
+                            if let Err(err) = channels.publish(TOPIC, text.as_str()) {{
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn-cli/src/generate/channel.rs:291]: This suggestion doesn't compile. `channels.publish` takes `impl Into&lt;ChannelMessage&gt;`, and `ChannelMessage` only implements `From&lt;String&gt;`/`From&lt;&str&gt;` — not `From&lt;&Utf8Bytes&gt;`. Deref coercion isn't applied when resolving a generic trait bound like this, so `&text` fails with `E0277: the trait bound ChannelMessage: From<&Utf8Bytes> is not satisfied` (verified locally). Keeping `.as_str()` here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn-cli/src/generate/channel.rs [hunk 2]
```diff
@@ -0,0 +1,1073 @@
...
+
+/// `POST /{snake_name}/messages` — publish a message to every connected client.
+#[post("/{snake_name}/messages")]
+pub async fn {snake_name}_publish(
+    State(state): State<AppState>,
+    Form(form): Form<{struct_name}Form>,
+) -> AutumnResult<&'static str> {{
+    state
+        .channels()
+        .publish(TOPIC, form.message.as_str())
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn-cli/src/generate/channel.rs:326]: This suggestion doesn't compile. `channels.publish`/`broadcast().publish` take `impl Into&lt;ChannelMessage&gt;` (or `BroadcastPayload`), and neither implements `From&lt;&String&gt;` — only `From&lt;String&gt;`/`From&lt;&str&gt;`. Deref coercion doesn't apply through a generic `impl Into&lt;T&gt;` bound, so `&form.message` fails with `E0277: the trait bound ChannelMessage: From<&String> is not satisfied` (verified locally). Keeping `.as_str()` here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: autumn-cli/src/generate/channel.rs [hunk 3]
```diff
@@ -0,0 +1,1073 @@
...
+    state
+        .broadcast()
+        .publish(TOPIC, fragment)
+        .map_err(AutumnError::internal_server_error)?;
+    Ok("published")"#
+        }
+        Transport::Ws => {
+            r#"    state
+        .channels()
+        .publish(TOPIC, form.message.as_str())
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn-cli/src/generate/channel.rs:363]: Same issue as the other two comments on this suggestion: `.publish(...)` takes `impl Into&lt;ChannelMessage&gt;`, which is only implemented for `String`/`&str`, not `&String` — deref coercion doesn't kick in through a generic bound, so this would fail with `E0277`. Keeping `.as_str()` here.

---
_Generated by [Claude Code](https://claude.ai/code)_
